packages/uploadthing/src/sdk/index.ts

```diff
@@ -55,8 +60,42 @@ export const getFileUrls = async (fileKeys: string[] | string) => {
     },
     body: JSON.stringify({ fileKeys }),
   });
+
   if (!res.ok) {
     throw new Error("Failed to get file urls");
   }
+
   return res.json().then(({ data }) => data as { key: string; url: string }[]);
 };
+
+/**
+ * Request file list from UploadThing storage.
+ *
+ * @example
+ * const data = await listFiles();
+ * console.log(data); // { key: "2e0fdb64-9957-4262-8e45-f372ba903ac8_image.jpg", id: "2e0fdb64-9957-4262-8e45-f372ba903ac8" }
+ */
+export const listFiles = async () => {
+  if (!UT_SECRET) throw new Error("Missing UPLOADTHING_SECRET env variable.");
+
+  // TODO: Implement filtering and pagination
+  const res = await fetch(generateUploadThingURL("/api/listFiles"), {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      "x-uploadthing-api-key": UT_SECRET,
+      "x-uploadthing-version": UPLOADTHING_VERSION,
+    },
+  });
+
+  const json = (await res.json()) as
+    | { files: { key: string; id: string }[] }
+    | { error: string };
```

Comment:
I saw that you also returned a `message` prop on the infra side. Which is more descriptive to use?

Reply:
Honestly not sure what will come out of the message, could be more descriptive, could be unintelligible garbage. 😅

Reply:
aight let's just go with this then